__tests__/accessiblity-audit.test.js

```diff
@@ -20,58 +16,47 @@ async function audit (page) {
     .exclude('.govuk-skip-link')
     // axe reports that the back to top button is not inside a landmark, which is intentional.
     .exclude('.app-back-to-top')
+    // axe reports that the frame "does not have a main landmark" and example <h1> headings
+    // violate "Heading levels should only increase by one", which is intentional.
+    // https://github.com/alphagov/govuk-design-system/pull/2442#issuecomment-1326600528
+    .exclude('.app-example__frame')
 
-  const results = await axe.analyze()
-
-  return results.violations
+  return axe.analyze()
 }
 
-beforeAll(async () => {
-  page = await setupPage()
-})
-
-afterAll(async () => {
-  await page.close()
-})
-
 describe('Accessibility Audit', () => {
   describe('Home page - layout.njk', () => {
     it('validates', async () => {
-      await page.goto(baseUrl + '/', { waitUntil: 'load' })
-      const violations = await audit(page)
-      expect(violations).toEqual([])
+      const results = await analyze(page, '/')
+      expect(results).toHaveNoViolations()
```

Comment:
I've used the [jest-axe](https://github.com/nickcolley/jest-axe#readme) `.toHaveNoViolations()` reporter for a nicer output

Reply:
I like this on the whole. Seems a bit weird to have to straddle the line between `jest-axe` and puppeteer, but I wholeheartedly support better error reporting, so a good change overall.

Reply:
Thanks @domoscargin

Yeah we do a similar thing on `govuk-frontend` where we [import the `jest-axe` reporter separately](https://github.com/alphagov/govuk-frontend/blob/main/src/govuk/components/accordion/template.test.js#L15)

```js
const results = await axe($.html())
expect(results).toHaveNoViolations()
```